RESTaurant/public/setupEventListeners.js

```diff
@@ -0,0 +1,29 @@
+function setupEventListeners(){
+
+    let elements = document.getElementsByName("deleteButton");
+
+    if (!elements) {
+        return
+    }
+
+    elements.forEach(el => {
+        const companyId = el.getAttribute("companyId")
+        el.addEventListener("click", () => {
+            deleteCompany(companyId)
+        })
+    })
+
+}
+
+async function deleteCompany(id){
+    const url = `/companies/${id}`;
+    await fetch(url, {
+        method: "DELETE",
+        headers: {
```

Comment:
don't think you need these headers since you're not sending anything in the body